packages/webdriverio/src/commands/element/scrollIntoView.ts

```diff
@@ -2,9 +2,127 @@ import logger from '@wdio/logger'
+import type { CustomScrollIntoViewOptions, MobileScrollIntoViewOptions } from 'src/types.js'
 
 const log = logger('webdriverio')
 
+type MobileScrollUntilVisibleOptions = {
+    browser: WebdriverIO.Browser;
+    element: WebdriverIO.Element;
+    maxScrolls: number;
+    scrollDirection: MobileScrollDirection;
+    scrollableElement: WebdriverIO.Element | null;
+};
+
+async function getScrollableElement({
+    browser,
+    options
+}: {
+    browser: WebdriverIO.Browser,
+    options?: MobileScrollIntoViewOptions
+    }): Promise<WebdriverIO.Element | null> {
+    if (options?.scrollableElement) {
+        return options?.scrollableElement
+    }
+    const defaultAndroidSelector = '//android.widget.ScrollView'
+    const defaultIosSelector = '-ios predicate string:type == "XCUIElementTypeApplication"'
+    const selector = browser.isIOS
+        ? // For iOS, we need to find the application element, if we can't find it, we should throw an error
+        defaultIosSelector
+        : // There is always a scrollview for Android or, if this fails we should throw an error
+        defaultAndroidSelector
+    // Not sure why we need to do this, but it seems to be necessary
+    const scrollableElements = (await browser.$$(
+        selector
+    )) as unknown as WebdriverIO.Element[]
+
+    if (scrollableElements.length > 0) {
+        return scrollableElements[0]
+    }
+
+    throw new Error(
+        `Default scrollable element '${browser.isIOS ? defaultIosSelector : defaultAndroidSelector}' not found.`
+    )
+}
+
+async function mobileScrollUntilVisible({
+    browser,
+    element,
+    scrollableElement,
+    maxScrolls,
+    scrollDirection,
+}: MobileScrollUntilVisibleOptions): Promise<{ hasScrolled: boolean; isVisible: boolean;  }> {
+    let isVisible = false
+    let hasScrolled = false
+    let scrolls = 0
+
+    while (!isVisible && scrolls < maxScrolls) {
+        try {
+            isVisible = await element.isDisplayed()
+        } catch {
+            isVisible = false
+        }
+
+        if (isVisible) {break}
+
+        if (browser.isIOS) {
+            await browser.execute('mobile: scroll', {
+                elementId: (await scrollableElement)?.elementId,
+                direction: scrollDirection,
+            })
+            hasScrolled = true
+        } else {
+            await browser.execute('mobile: scrollGesture', {
+                elementId: (await scrollableElement)?.elementId,
+                direction: scrollDirection,
+                percent: 0.5,
+            })
+            hasScrolled = true
+        }
+
+        scrolls++
+    }
+
+    return { hasScrolled, isVisible }
+}
+
+async function nativeMobileScrollIntoView({
+    browser,
+    element,
+    options
+}: {
+    browser: WebdriverIO.Browser,
+    element: WebdriverIO.Element,
+    options: MobileScrollIntoViewOptions
+    }) {
+    const defaultOptions = {
+        maxScrolls: 10,
+        scrollDirection: MobileScrollDirection.Down,
```

Comment:
What if we already scrolled all the way down and now have to do scroll up to find the element?

Reply:
I'll update the error message if we can't find an element. I believe that users need to know their app in which direction to swipe to find the element. Sadly, native app automation is not that easy in comparison to web 😅